component/attr_cache/attr_cache.go

```diff
@@ -802,7 +775,13 @@ func (ac *AttrCache) CopyFromFile(options internal.CopyFromFileOptions) error {
 		}
 		// TODO: Could we just update the size and mod time of the file here? Or can other attributes change here?
 		// TODO: we're RLocking the cache but we need to also lock this attr item because another thread could be reading this attr item
-		ac.invalidatePath(options.Name)
+
+		toBeInvalid, err := ac.cacheMap.get(options.Name) //empty for TestCopyFromFileDoesNotExist()
+		if err != nil {
+			log.Err("AttrCache::CopyFromFile : The attribute item could not be invalidated in the cache due to the following error: ", err)
+		} else {
+			toBeInvalid.invalidate()
```

Comment:
We chatted about whether the cache should invalidate modified files. I think we decided that it should not, but that it could be a separate PR.
Your new `renameCachedDirectory()` preserves cache entries for moved files. If you'd like to not expand your PR scope, and leave this as is here, I think that's fine. It'll create a mixed implementation, where files sometimes keep their cache entries, and other times get invalidated. I think we can tie that up in a subsequent PR.

Reply:
I used the moveAttrCacheItem() to reinsert the modified file and it is no longer bein invalidated.